### PR TITLE
docs: change log to ln in Softplus function and class

### PR DESCRIPTION
**Pull request by Serenazhu**
Updated the math formula in the softplus function in torch.nn.functional.py and the Softplus class in torch.nn.modules.activation.py from log to ln for correctness and accuracy.

### CODE REVIEW

**Review by mikaylagawarecki — APPROVED** [review]
Checks out with the use of log1p, thanks and sorry for the delay, we can merge this if you sign the CLA in https://github.com/pytorch/pytorch/pull/146199#issuecomment-2628372918


https://github.com/pytorch/pytorch/blob/316c188a5e700c276ed2c4ec1e15e5b4eee6d1a8/aten/src/ATen/native/cpu/Activation.cpp#L962
